themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/signingin-page/SigningInPage.tsx

```diff
@@ -0,0 +1,183 @@
+            .then((response: AxiosResponse<PasswordDetails>) => {
+                if (response.data.lastUpdate) {
+                    const lastUpdate: number = response.data.lastUpdate;
+                    this.setState({lastPasswordUpdate: lastUpdate});
+                }
+            });
+    }
+    
+    private handleTwoFactorSwitch = () => {
+        if (this.state.twoFactorEnabled) {
+            this.setState({twoFactorEnabled: false, twoFactorEnabledText: Msg.localize('twoFactorDisabled')})
+        } else {
+            this.setState({twoFactorEnabled: true, twoFactorEnabledText:  Msg.localize('twoFactorEnabled')})
+        }
+    }
+
+    private handleRemoveTOTP = () => {
+      AccountServiceClient.Instance.doDelete("/totp/remove")
+        .then(() => {
+            this.setState({isTotpConfigured: false});
+            ContentAlert.success('successTotpRemovedMessage');
+        });
+    }
+
+    public render(): React.ReactNode {
+        let lastPwdUpdate: string = Msg.localize('unknown');
+        if (this.state.lastPasswordUpdate) {
+            lastPwdUpdate = moment(this.state.lastPasswordUpdate).format('LLL');
+        }
+
+        return (
+            <ContentPage title="signingIn" 
+                     introMessage="signingInSubMessage">
+                <Stack gutter='md'>
+                    <StackItem isFilled>
+                        <Title headingLevel={TitleLevel.h2} size='2xl'>
+                        <strong><Msg msgKey='password'/></strong>
+                        </Title>
+                        <DataList aria-label='foo'>
+                            <DataListItemRow>
+                                <DataListItemCells
+                                    dataListCells={[
+                                        <DataListCell key='password'><Msg msgKey='password'/></DataListCell>,
+                                        <DataListCell key='lastPwdUpdate'><strong><Msg msgKey='lastUpdate'/>: </strong>{lastPwdUpdate}</DataListCell>,
+                                        <DataListCell key='spacer'/>
+                                    ]}/>
+                                <DataListAction aria-labelledby='foo' aria-label='foo action' id='setPasswordAction'>
+                                    <Button variant='primary'onClick={()=> this.updatePassword.execute()}><Msg msgKey='update'/></Button>
+                                </DataListAction>
+                            </DataListItemRow>
+                        </DataList>
+                    </StackItem>
+                    <StackItem isFilled>
+                        <Title headingLevel={TitleLevel.h2} size='2xl'>
+                        <strong><Msg msgKey='twoFactorAuth'/></strong>
+                        </Title>
+                        <DataList aria-label='foo'>
+                            <DataListItemRow>
+                                <DataListAction aria-labelledby='foo' aria-label='foo action' id='twoFactorOnOff'>
```

Comment:
```
aria-labelledby='foo' aria-label='foo action'
```
Should there be something more relevant? :)

Reply:
Yes.  But we won't address accessibility until after tech preview.  It needs to be done for the entire app.  I think there is already a JIRA for it.  If not, feel free to create one.

Reply:
Created https://issues.jboss.org/browse/KEYCLOAK-12248